internal/generate/crd/crd.go

```diff
@@ -0,0 +1,342 @@
+		return nil, err
+	}
+	cache := genutil.GetCache()
+	infos, err := afero.ReadDir(cache, cacheOutputDir)
+	if err != nil {
+		return nil, err
+	}
+	for _, info := range infos {
+		if info.IsDir() {
+			continue
+		}
+		path := filepath.Join(cacheOutputDir, info.Name())
+		b, err := afero.ReadFile(cache, path)
+		if err != nil {
+			return nil, err
+		}
+		scanner := yamlutil.NewYAMLScanner(b)
+		modifiedCRD := []byte{}
+		for scanner.Scan() {
+			sb := scanner.Bytes()
+			typeMeta, err := k8sutil.GetTypeMetaFromBytes(sb)
+			if err != nil {
+				return nil, err
+			}
+			if typeMeta.Kind != "CustomResourceDefinition" {
+				continue
+			}
+			crd := &apiextv1beta1.CustomResourceDefinition{}
+			if err = yaml.Unmarshal(sb, crd); err != nil {
+				return nil, err
+			}
+			// controller-tools inserts an annotation and assumes that the binary
+			// that creates the CRD is controller-gen. In this case, we don't use
+			// controller-gen. Instead, we vendor and use the same library that
+			// controller-gen does.
+			//
+			// The value that gets populated in the annotation is based on the
+			// build info of the compiled binary, not on the version of the
+			// vendored controller-tools library.
+			//
+			// See: https://github.com/kubernetes-sigs/controller-tools/issues/348
+			//
+			// TODO(joelanford): Sort out what to do with this. Until then, let's
+			// just remove it.
+			delete(crd.Annotations, "controller-gen.kubebuilder.io/version")
+			if sb, err = k8sutil.GetObjectBytes(crd, yaml.Marshal); err != nil {
+				return nil, err
+			}
+			modifiedCRD = yamlutil.CombineManifests(modifiedCRD, sb)
+		}
+		if err = scanner.Err(); err != nil {
+			return nil, err
+		}
+		if len(modifiedCRD) != 0 {
+			fileNameNoExt := strings.TrimSuffix(info.Name(), filepath.Ext(info.Name()))
+			fileMap[fileNameNoExt+"_crd.yaml"] = modifiedCRD
+		}
+	}
+	return fileMap, nil
+}
+
+// generateNonGo generates a CRD for non-Go projects using a resource.
+func (g crdGenerator) generateNonGo() (map[string][]byte, error) {
+	crd := &apiextv1beta1.CustomResourceDefinition{}
+	fileMap := map[string][]byte{}
+	fileName := getFileNameForResource(g.resource)
+	path := filepath.Join(g.InputDir, fileName)
+	if _, err := os.Stat(path); err == nil {
+		b, err := ioutil.ReadFile(path)
+		if err != nil {
+			return nil, err
+		}
+		if err = yaml.Unmarshal(b, crd); err != nil {
+			return nil, err
+		}
+		// If version is new, append it to spec.versions.
+		hasVersion := false
+		for _, version := range crd.Spec.Versions {
+			if version.Name == g.resource.Version {
+				hasVersion = true
+				break
+			}
+		}
+		if !hasVersion {
+			// Let either the user or below logic determine whether this new
+			// version is stored or not.
+			crd.Spec.Versions = append(crd.Spec.Versions, apiextv1beta1.CustomResourceDefinitionVersion{
+				Name:    g.resource.Version,
+				Storage: false,
+				Served:  true,
+			})
+		}
+	} else if os.IsNotExist(err) {
```

Comment:
I think would be easier to read/understand as:
if err ;= call() ;err!= nil{
   //todo
}
if err ;= call() ;err!= nil{
   //todo
}
if err ;= call() ;err!= nil{
   //todo
}

then use the else

Reply:
You mean like
```go
_, err := os.Stat(path)
if err != nil && !os.IsNotExist(err) {
	...
}
if err != nil && os.IsNotExist(err) {
	...
}
// else
...
```